pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1214,6 +1216,38 @@ func (l *LibvirtDomainManager) UnfreezeVMI(vmi *v1.VirtualMachineInstance) error
 	return nil
 }
 
+func (l *LibvirtDomainManager) SoftRebootVMI(vmi *v1.VirtualMachineInstance) error {
+	domainRebootFlagValues := libvirt.DOMAIN_REBOOT_GUEST_AGENT
+	condManager := controller.NewVirtualMachineInstanceConditionManager()
+	if !condManager.HasConditionWithStatus(vmi, v1.VirtualMachineInstanceAgentConnected, k8sv1.ConditionTrue) {
+		if *vmi.Spec.Domain.Features.ACPI.Enabled {
```

Comment:
Probably same null-check is needed here: https://github.com/kubevirt/kubevirt/pull/6670/files#r748085148